compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2488,6 +2489,11 @@ impl<'tcx> SelectionContext<'_, 'tcx> {
         let InferOk { obligations, .. } = self
             .infcx
             .at(&cause, obligation.param_env)
+            // #116652
```

Comment:
Instead of linking all of these issues with `#116652`, can you just put `FIXME(DefineOpaqueTypes):`? That may be easier.